dm/master/server.go

```diff
@@ -1652,89 +1645,87 @@ func (s *Server) CheckTask(ctx context.Context, req *pb.CheckTaskRequest) (*pb.C
 	}
 	fromDB, err := conn.DefaultDBProvider.Apply(*dbConfig)
 	if err != nil {
-		return makeMysqlTaskResponse(err)
+		return makeMysqlWorkerResponse(err)
+	}
+	if err = cfg.Adjust(fromDB.DB); err != nil {
+		return makeMysqlWorkerResponse(err)
 	}
-	if err := cfg.Adjust(fromDB.DB); err != nil {
-		return makeMysqlTaskResponse(err)
+	if req.Config, err = cfg.Toml(); err != nil {
+		return makeMysqlWorkerResponse(err)
 	}
-	var resp *pb.MysqlTaskResponse
+	var resp *pb.MysqlWorkerResponse
 	if req.Op == pb.WorkerOp_StartWorker {
 		w := s.coordinator.GetWorkerBySourceID(cfg.SourceID)
 		if w != nil {
-			return &pb.MysqlTaskResponse{
+			return &pb.MysqlWorkerResponse{
 				Result: false,
 				Msg:    "Create worker failed. worker has been started",
 			}, nil
 		}
-		w, err := s.coordinator.AcquireWorkerForSource(cfg)
+		w, err := s.coordinator.AcquireWorkerForSource(cfg.SourceID)
 		if err != nil {
-			return &pb.MysqlTaskResponse{
-				Result: false,
-				Msg:    err.Error(),
-			}, nil
+			return makeMysqlWorkerResponse(err)
 		}
-		resp, err = w.OperateMysqlTask(ctx, req, s.cfg.RPCTimeout)
+
+		resp, err = w.OperateMysqlWorker(ctx, req, s.cfg.RPCTimeout)
 		if err != nil {
 			// TODO: handle error or backoff
 			s.coordinator.HandleStartedWorker(w, cfg, false)
-			return &pb.MysqlTaskResponse{
-				Result: false,
-				Msg:    errors.ErrorStack(err),
-			}, nil
+			return makeMysqlWorkerResponse(err)
 		}
 		// TODO: handle error or backoff
 		s.coordinator.HandleStartedWorker(w, cfg, true)
 	} else if req.Op == pb.WorkerOp_UpdateConfig {
 		w := s.coordinator.GetWorkerBySourceID(cfg.SourceID)
 		if w == nil {
-			return &pb.MysqlTaskResponse{
+			return &pb.MysqlWorkerResponse{
 				Result: false,
 				Msg:    "Update worker config failed. worker has not been started",
 			}, nil
 		}
-		if resp, err = w.OperateMysqlTask(ctx, req, s.cfg.RPCTimeout); err != nil {
-			return &pb.MysqlTaskResponse{
-				Result: false,
-				Msg:    errors.ErrorStack(err),
-			}, nil
+		if resp, err = w.OperateMysqlWorker(ctx, req, s.cfg.RPCTimeout); err != nil {
+			return makeMysqlWorkerResponse(err)
 		}
 	} else {
 		w := s.coordinator.GetWorkerBySourceID(cfg.SourceID)
 		if w == nil {
-			return &pb.MysqlTaskResponse{
-				Result: false,
-				Msg:    "Stop worker failed. worker has not been started",
-			}, nil
-		}
-		if resp, err = w.OperateMysqlTask(ctx, req, s.cfg.RPCTimeout); err != nil {
-			return &pb.MysqlTaskResponse{
-				Result: false,
-				Msg:    errors.ErrorStack(err),
-			}, nil
+			if !s.coordinator.HandleStoppedWorker(nil, cfg) {
+				return &pb.MysqlWorkerResponse{
+					Result: false,
+					Msg:    "Stop worker failed. worker has not been started",
+				}, nil
+			}
+		} else {
+			if resp, err = w.OperateMysqlWorker(ctx, req, s.cfg.RPCTimeout); err != nil {
+				return &pb.MysqlWorkerResponse{
+					Result: false,
+					Msg:    errors.ErrorStack(err),
+				}, nil
+			}
+			s.coordinator.HandleStoppedWorker(w, cfg)
```

Comment:
Do not need to handle the returned result here?

Reply:
No, because it must be success